src/drivers/actuators/vertiq_io/vertiq_io.cpp

```diff
@@ -0,0 +1,306 @@
+	perf_begin(_loop_perf);
+	perf_count(_loop_interval_perf);
+
+	//If we should leave, then clean up our mess and get out
+	if (should_exit()) {
+		ScheduleClear();
+		exit_and_cleanup();
+		return;
+	}
+
+	// telemetry device update request?
+	//This happens whenever the module is started
+	if (_request_telemetry_init.load()) {
+		_serial_interface.init_serial(_telemetry_device, _param_vertiq_baud.get());
+		_request_telemetry_init.store(false);
+	}
+
+	//Handle IQUART reception and transmission
+	_client_manager.HandleClientCommunication();
+
+	//If we're supposed to ask for telemetry from someone
+	if (_telem_bitmask) {
+		//Grab the next ID, and if it's real get it ready to send out
+		uint16_t next_telem = _telem_manager.UpdateTelemetry();
+
+		if (next_telem != _impossible_module_id) {
+			_telemetry_request_id = next_telem;
+		}
+	}
+
+	//Get the most up to date version of our parameters/check if they've changed
+	parameters_update();
+
+	//Make sure we also update the mixing output to get the most up to date configuration
+	_mixing_output.update();
+	_mixing_output.updateSubscriptions(true);
+
+	//Go ahead and check to see if our actuator test has gotten anything new
+	if (_actuator_test_sub.updated()) {
+		_actuator_test_sub.copy(&_actuator_test);
+
+		//Our test is active if anyone is giving us commands through the actuator test
+		_actuator_test_active = _actuator_test.action == actuator_test_s::ACTION_DO_CONTROL;
+	}
+
+	//stop our timer
+	perf_end(_loop_perf);
+}
+
+
+void VertiqIo::parameters_update()
+{
+	//If someone has changed any parameter in our module. Checked at 1Hz
+	if (_parameter_update_sub.updated()) {
+		//Grab the changed parameter with copy (which lowers the "changed" flag)
+		parameter_update_s param_update;
+		_parameter_update_sub.copy(&param_update);
+
+		// If any parameter updated, call updateParams() to check if
+		// this class attributes need updating (and do so).
+		updateParams();
+
+		//If the target module ID changed, we need to update all of our parameters
+		if (_param_vertiq_target_module_id.get() != _client_manager.GetObjectIdNow()) {
+			_client_manager.UpdateClientsToNewObjId(_param_vertiq_target_module_id.get());
+
+			//Make sure we start a new read!
+			_param_vertiq_trigger_read.set(true);
+			_param_vertiq_trigger_read.commit_no_notification();
+		}
+
+		//If you're set to re-read from the motor, mark all of the IQUART parameters for reinitialization, reset the trigger, and then update the IFCI params
+		if (_param_vertiq_trigger_read.get()) {
+			_client_manager.MarkIquartConfigsForRefresh();
+			_param_vertiq_trigger_read.set(false);
+			_param_vertiq_trigger_read.commit_no_notification();
+		}
+
+		_client_manager.UpdateIquartConfigParams();
+	}
+}
+
+bool VertiqIo::updateOutputs(bool stop_motors, uint16_t outputs[MAX_ACTUATORS], unsigned num_outputs,
+			     unsigned num_control_groups_updated)
+{
+#ifdef CONFIG_USE_IFCI_CONFIGURATION
+
+	if (_mixing_output.armed().armed) {
```

Comment:
if the _flight controller_ is armed check if we're supposed to force the motor to arm. Generally just spit out the broadcast message with the output throttles coming in from outputs.